### ISSUE TITLE
Construct path for `/.well-known/openid-configuration`

### ISSUE THREAD

**Issue by trengrj** [semi-technologies/weaviate#2303]
Currently the well-known openid configuration endpoint constructs the url using Sprintf

https://github.com/semi-technologies/weaviate/blob/8ba795c5dbc1e2039ef3fa1bf53a023c0fd1a58e/adapters/handlers/rest/handlers_misc.go#L61

This causes issues if the issuer has a trailing slash (which is present in Auth0). Recommend to change from Sprintf to use `net/url` or similar to correctly handle trailing slashes i.e.:

```golang
target, err := url.JoinPath(serverConfig.Config.Authentication.OIDC.Issuer, "/.well-known/openid-configuration")
```

**Comment by etiennedi** [comment]
Great catch!

**Comment by etiennedi** [comment]
Released in `v1.16`.
